frontend/src/components/AddPermissions/UserNameSection.vue

```diff
@@ -0,0 +1,208 @@
+<script setup lang="ts">
+import { ref, watch } from "vue";
+import SearchLocateIcon from "@carbon/icons-vue/es/search--locate/16";
+import Button from "@/components/UI/Button.vue";
+import UserIdentityCard from "@/components/AddPermissions/UserIdentityCard.vue";
+import useAuth from "@/composables/useAuth";
+import { IdpProvider } from "@/enum/IdpEnum";
+import { AppActlApiService } from "@/services/ApiServiceFactory";
+import { useMutation } from "@tanstack/vue-query";
+import type {
+    IdimProxyBceidInfoSchema,
+    IdimProxyIdirInfoSchema,
+} from "fam-app-acsctl-api";
+import { UserType } from "fam-app-acsctl-api";
+import InputText from "primevue/inputtext";
+import { ErrorMessage, Field } from "vee-validate";
+
+const auth = useAuth();
+
+const props = withDefaults(
+    defineProps<{
+        domain: UserType;
+        user: IdimProxyIdirInfoSchema | IdimProxyBceidInfoSchema | null;
+        appId: number;
+        fieldId?: string;
+        helperText?: string;
+    }>(),
+    {
+        fieldId: "user",
+    }
+);
+
+const PERMISSION_REQUIRED_FOR_OPERATION = "permission_required_for_operation";
+
+const emit = defineEmits(["setVerifyResult"]);
+
+const userIdInput = ref<string>("");
+
+const errorMsg = ref("");
+
+/**
+ * Checks if the provided user ID matches the currently logged-in user's ID.
+ */
+const isCurrentUser = (): boolean => {
+    const userId = userIdInput.value?.toLowerCase();
+    const loggedInUserId = auth.authState.famLoginUser?.username?.toLowerCase();
+
+    return userId === loggedInUserId;
+};
+
+const handleMutationError = (error: any) => {
+    emit("setVerifyResult", null);
+
+    // Check if error and response properties exist
+    if (error.response && error.response.status === 403) {
+        const detail = error.response.data?.detail;
+        if (detail?.code === PERMISSION_REQUIRED_FOR_OPERATION) {
+            errorMsg.value = `${detail.description}. Org name: ${
+                auth.authState.famLoginUser?.organization ??
+                "Unknown organization"
+            }`;
+        } else {
+            errorMsg.value = "An unknown error occurred.";
+        }
+    } else {
+        errorMsg.value =
+            "Unable to verify the user due to a network or server error.";
+    }
+};
+
+const verifyIdirMutation = useMutation({
```

Comment:
I chose useMutation here because idirSearch is a verification request that depends on a changing user ID, which needs to be triggered manually rather than passively cached. useQuery is typically better for caching static or infrequent data, while useMutation gives us more control for dynamic or ad-hoc requests like this verification process. Although it’s a GET request, the use of useMutation fits here to handle the changing input and avoid unnecessary caching.

Reply:
Got it. Maybe `useQuery` need some new option.